src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -86,6 +88,16 @@ protected function instantiateController($class)
             return $this->container->get($class);
         }
 
-        return parent::instantiateController($class);
+        $e = null;
+
+        try {
+            return parent::instantiateController($class);
+        } catch (FatalThrowableError $e) {
+        } catch (\ArgumentCountError $e) {
+        }
+
+        if (null !== $e && $this->container instanceof Container && in_array($class, $this->container->getRemovedIds())) {
+            throw new \RuntimeException(sprintf('The controller %s seems to be removed from the container, which means that he maybe private, add the tag \'controller.service_arguments\' or made it public.', $class), 0, $e);
```

Comment:
I propose this:
`new \LogicException(sprintf('Controller "%s" cannot be fetched from the container because it is not public. Did you forget to tag the service with "controller.service_arguments"?', $class), 0, $e);`